hazelcast/src/main/java/com/hazelcast/jet/impl/JetInstanceImpl.java

```diff
@@ -61,11 +64,34 @@ public class JetInstanceImpl extends AbstractJetInstance<Address> {
         this.config = config;
     }
 
-    @Nonnull @Override
+    @Nonnull
+    @Override
     public JetConfig getConfig() {
         return config;
     }
 
+    @Override
+    public void uploadJob(@Nonnull Path jarPath, String snapshotName, String jobName, String mainClass,
+                          List<String> jobParameters) {
+
+        try {
+            JobMetaDataParameterObject parameterObject = new JobMetaDataParameterObject();
+            parameterObject.setSnapshotName(snapshotName);
+            parameterObject.setJobName(jobName);
+            parameterObject.setMainClass(mainClass);
+            parameterObject.setJobParameters(jobParameters);
+            parameterObject.setJarPath(jarPath);
+
+            JetServiceBackend jetServiceBackend = nodeEngine.getService(JetServiceBackend.SERVICE_NAME);
+
+            jetServiceBackend.checkIfCanRunJar();
+            jetServiceBackend.runJar(parameterObject);
```

Comment:
This line is actually not run in any of the new tests.

Reply:
Added new test